h2o-algos/src/main/java/hex/modelselection/ModelSelectionUtils.java

```diff
@@ -1130,15 +1132,17 @@ public static PredNameMinZVal findCatMinZVal(GLMModel model, List<Double> zValLi
                     }
                     if (catZValues.size() > 0) {
                         double oneCatMinZ = catZValues.stream().max(Double::compare).get(); // choose the best z-value here
-                        if (oneCatMinZ < minCatVal) {
-                            minCatVal = oneCatMinZ;
-                            catPredMinZ = columnNames[catInd];
-                        }
+                        bestZValues.add(oneCatMinZ);
+                        catPredNames.add(columnNames[catInd]);
                     }
                 }
             }
         }
-        return new PredNameMinZVal(catPredMinZ, minCatVal);
+        if (bestZValues.size() < 1)
+            return null;
+        double maxCatLevel = bestZValues.stream().min(Double::compare).get();
```

Comment:
I would rename this method to something like `findCatMinOfMaxZVal` so you reflect that the behavior is actually selecting minimum of maximum Z values.